test/Factory/Factory.test.ts

```diff
@@ -0,0 +1,120 @@
+import { ethers } from 'hardhat';
+import { expect } from 'chai';
+import { Contract, Signer } from 'ethers';
+import { Factory, MyToken } from '../../typechain-types'; // Adjust the path according to your project structure
+
+
+
+describe('Factory Contract Tests', () => {
+
+    // Factory variables
+    let factory: Factory;
+    let signers: Signer[];
+    let admin: Signer;
+    let updater: Signer;
+    let mockAddress: Signer;
+    let implementationAddress: string;
+    let implementation2Address: string;
+    let encodedData: string;
+
+    //SuperQuorumGovernor Variables
+    let executionDelay = 0;
+    const votingDelay = 2;
+    const votingPeriod = 5; // 5 blocks
+    const extension = 0;
+    const quorumFraction = 10;
+    const superQuorumFraction = 60; // 60%
+    const proposalThreshold = 0;
+    const name = "MyGovernor"
+
+    // Fixture to deploy the contract and set initial state
+    async function deployFactoryFixture() {
+        signers = await ethers.getSigners();
+        [admin, updater, mockAddress] = signers;
+
+        const FactoryContract = await ethers.getContractFactory('Factory');
+        factory = await FactoryContract.deploy() as Factory;
+
+        // Deploy a mock implementation contract and set it in the factory
+        const ImplementationContract = await ethers.getContractFactory('SuperQuorumGovernorUpgradeable');
+        const token_factory = await ethers.getContractFactory("MyToken");
+        const timelock_factory = await ethers.getContractFactory("TimelockUpgradeable")
+        const token = await token_factory.deploy(admin.address);
+        const timelock = await timelock_factory.deploy();
+        await timelock.initialize(0, [], [], admin.address);
+
+
+        const implementation = await ImplementationContract.deploy();
+        const implementation2 = await ImplementationContract.deploy();
+
+        implementationAddress = await implementation.getAddress();
+        implementation2Address = await implementation2.getAddress();
```

Comment:
we don't need two implementation contracts.
of the same govenor.

we just use cloneAndInitialize twice.

Reply:
I use it in line 95 to confirm that we're able to update the implementation to a new address. It's just a convenience function.